app/assets/javascripts/components/dashboard/navigation/content_panel/content_panel.jsx

```diff
@@ -1,8 +1,46 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import Sidebar from './sidebar.jsx';
 import MenuContent from './menu_content.jsx';
+import _ from 'lodash';
 
-export default class ContentPanel extends Component {
+
+function getSignal(ownProps, state) {
+  const { route, params } = ownProps;
+
+  if (route.path === 'new') {
+    return {
+      type: params.type,
+    }
+  } else {
+    const signals = state.models.listenSignals.data;
+    const key = _.findKey(signals, { id: parseInt(params.id) });
+    const signal = signals[key];
+
+    const responses = (signal) => {
+      const data = [];
+
+      _.forEach(signal.responses, function(key) {
+        data.push(state.models.responses.data[key]);
+      });
+      return data;
+    };
+
+    const newSignal = (signal, responses) => {
```

Comment:
Let's not define the function withing this function. We can define it outisde of it.